### PR TITLE
Add basic logging mechanism

### PR DESCRIPTION
**Pull request by netoarmando**
Configure a logger to be used in the whole application.

---

Based on the table [here](https://docs.python.org/3/howto/logging.html#when-to-use-logging, I wonder if we should keep all the prints for stdout and stderr messages instead of replacing them.

### CODE REVIEW

**Comment by pvoborni** [comment]
It will probably depend on the action to run.  Actions like 'up' and 'destroy' should probably log almost everything. They will be used in CIs so the log should be available. I think that we can log more info (e.g. info from providers) to DEBUG level than we now print to console. 

For human 'console' actions like 'list' and 'ssh' we probably will mostly log to INFO and print.

**Comment by netoarmando** [comment]
> It will probably depend on the action to run. Actions like 'up' and 'destroy' should probably log almost everything. They will be used in CIs so the log should be available. I think that we can log more info (e.g. info from providers) to DEBUG level than we now print to console.
> 
> For human 'console' actions like 'list' and 'ssh' we probably will mostly log to INFO and print.

Agreed! Let me enhance this.

**Comment by pvoborni** [comment]
I see that there is used pattern:
```
        logger.info(f"Hosts to delete: {names}")
        print(f"Hosts to delete: {names}")
```
Do we want this code duplicity with strings?  

In the previous comment I thought that we will:
* log INFO level into console and DEBUG level into file implicitly. 
* log DEBUG into console when --debug is passed
* have separate `print()` statements only for text which we don't want to log into file.

I'm also open to not have the console logger but in such case we might want to have some utility function like
```
def print_log(logger, msg, level="INFO"):    
    file = sys.stdout
    if level in ("ERROR", "CRITICAL"):
       file = syst.stderr
    print(msg, file=file)
    logger.log(level, msg)
```

**Comment by netoarmando** [comment]
@pvoborni it was hard to feel comfortable with the idea of logging+printing a CLI application. Specially because the common practice is to log to stdout. But I think your points now made it clear to me. 

About the duplication, I wonder if most of the strings would be the same or not. However, I guess all of this is fixed by having a custom `print` function. It looks strange, but I think is the best solution.

**Comment by netoarmando** [comment]
Earlier today @pvoborni and I talked about this and we agreed to the following points:
- Remove all `print` and replace them with `logger.info`
- One handler would output the messages to the console without timestamps
- Another handler will display more info, such as the timestamps

File: src/aiohabit/providers/openstack.py [hunk 1]
```diff
@@ -109,7 +113,7 @@ async def init(self, image_names=None):
         self.limits = limits
         object_end = datetime.now()
         object_duration = object_end - object_start
-        print(f"Open Stack environment objects " f"load duration: {object_duration}")
+        logger.info(f"Open Stack environment objects " f"load duration: {object_duration}")
```
Comment (Tiboris) [inline src/aiohabit/providers/openstack.py:116]: Line is too long

File: src/aiohabit/providers/openstack.py [hunk 2]
```diff
@@ -312,8 +316,8 @@ async def can_provision(self, reqs):
         req_vcpus = used_vcpus + vcpus
         req_memory = used_memory + ram
 
-        print(f"Required vcpus: {vcpus}, " f"used: {used_vcpus}, max: {limit_vcpus}")
-        print(f"Required ram: {ram}, used: {used_memory}, max: {limit_memory}")
+        logger.info(f"Required vcpus: {vcpus}, " f"used: {used_vcpus}, max: {limit_vcpus}")
```
Comment (Tiboris) [inline src/aiohabit/providers/openstack.py:319]: Line is too long

**Comment by Tiboris** [comment]
@pvoborni  Let us merge this as well. Thanks @netoarmando

**Pull request merged**